frontend/app/actions/RegistrationUri.scala

```diff
@@ -0,0 +1,51 @@
+package actions
+
+import com.gu.membership.salesforce.Tier
+import com.netaporter.uri.Uri
+import configuration.Config
+import play.api.mvc.RequestHeader
+
+object RegistrationUri {
+  val REFERRER_HEADER: String = "referer"
+  val CAMPAIGN_SOURCE: String = "MEM"
+
+  def parse(request: RequestHeader) = {
+    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+    extractCampaignCode(request.headers(REFERRER_HEADER), request.path) match {
```

Comment:
This has caused a lot of errors on the site, see:

https://app.getsentry.com/the-guardian/membership/group/90393824
https://app.getsentry.com/the-guardian/membership/group/90393829